account/user.go

```diff
@@ -67,19 +68,36 @@ func (m *GormUserRepository) Load(ctx context.Context, id uuid.UUID) (*User, err
 	return &native, err
 }
 
+// loadActiveUserByEmail returns a single User with given email and deletedAt != NULL
+func (m *GormUserRepository) loadActiveUserByEmail(ctx context.Context, email string) *User {
+	defer goa.MeasureSince([]string{"goa", "db", "user", "load"}, time.Now())
+
+	var native User
+	err := m.db.Table(m.TableName()).Where("email = ? AND deleted_at IS NULL", email).First(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil
+	}
+	return &native
+}
+
 // Create creates a new record.
 func (m *GormUserRepository) Create(ctx context.Context, u *User) error {
 	defer goa.MeasureSince([]string{"goa", "db", "user", "create"}, time.Now())
 
-	u.ID = uuid.NewV4()
+	user := m.loadActiveUserByEmail(ctx, u.Email)
+	if user == nil {
+		// User does not exist hence can be created safely
+		u.ID = uuid.NewV4()
 
-	err := m.db.Create(u).Error
-	if err != nil {
-		goa.LogError(ctx, "error adding User", "error", err.Error())
-		return err
-	}
+		err := m.db.Create(u).Error
+		if err != nil {
+			goa.LogError(ctx, "error adding User", "error", err.Error())
+			return err
+		}
 
-	return nil
+		return nil
+	}
+	return errors.New("Record already exist")
```

Comment:
It is better to use error object implemented the `error` interface: https://golang.org/pkg/builtin/#error